File: 백준/Silver/11060. 점프 점프/점프 점프.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int[] num = new int[n + 1];
        long[] dp = new long[1101];
        String[] input = br.readLine().split(" ");
        for (int i = 1; i <= n; i++) {
            num[i] = Integer.parseInt(input[i - 1]);
            dp[i] = Integer.MAX_VALUE;
        }

        dp[1] = 0;
        for (int i = 1; i <= n; i++) {
            if (dp[i] >= Integer.MAX_VALUE) {
                continue;
            }

            for (int j = 1; j <= num[i]; j++) {
                dp[i + j] = Math.min(dp[i + j], dp[i] + 1);
            }
        }

        System.out.println(dp[n] >= Integer.MAX_VALUE ? -1 : dp[n]);
    }
}

File: 백준/Silver/11060. 점프 점프/점프 점프.js
let [n, input] = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "input.txt")
  .toString()
  .trim()
  .split("\n");

let a = input.split(" ").map(Number);
let dp = Array(+n).fill(Number.MAX_SAFE_INTEGER);

dp[0] = 0;

for (let i = 0; i < +n; i++) {
  if (dp[i] === Number.MAX_SAFE_INTEGER || a[i] === 0) {
    continue;
  }

  for (let j = 1; j <= a[i]; j++) {
    if (i + j < +n) {
      dp[i + j] = Math.min(dp[i + j], dp[i] + 1);
    }
  }
}

console.log(dp[n - 1] === Number.MAX_SAFE_INTEGER ? -1 : dp[n - 1]);
